types/xdg_shell_v6/wlr_xdg_toplevel_v6.c

```diff
@@ -19,6 +19,11 @@ void destroy_xdg_toplevel_v6(struct wlr_xdg_surface_v6 *surface) {
 	assert(surface->role == WLR_XDG_SURFACE_V6_ROLE_TOPLEVEL);
 	unmap_xdg_surface_v6(surface);
 
+	if (surface->toplevel->client_pending.fullscreen_output) {
+		struct wlr_xdg_toplevel_v6_state *client_pending =
+			&surface->toplevel->client_pending;
+		wl_list_remove(&client_pending->fullscreen_output_destroy.link);
```

Comment:
We should probably also reset `fullscreen_output` to NULL here.

Reply:
I didn't set it to NULL since the toplevel is being freed right after the if anyway. I can add it if you want though.

Reply:
Oh, no, you're right.